### PR TITLE
Transition to GHA; replace linkchecker (SOFTWARE- 4343)

### PR DESCRIPTION
**Pull request by brianhlin**
Documentation to come but I'm thinking we can take the new workflows and share them: https://docs.github.com/en/free-pro-team@latest/actions/learn-github-actions/sharing-workflows-with-your-organization

So updates would look something like:

1. If the user isn't comfortable doing this themselves, reach out to us at help@
1. Use Mat's link fixer upper for new MkDocs versions
1. `git rm ci; git rm .travis.env .travis.yml deploy*.enc deploy*.pub; git commit -m "Disable Travis CI"`
1. `git push origin master`
1. Set up validate and deploy workflows through GitHub UI

### CODE REVIEW

File: .github/workflows/validate-mkdocs.yml [hunk 1]
```diff
@@ -0,0 +1,29 @@
+name: Valdate static MkDocs pages
+on: pull_request
+
+jobs:
+  validate-mkdocs:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
```
Comment (matyasselmeci) [inline .github/workflows/validate-mkdocs.yml:10]: What does fetching with a depth of 0 do? It seems nonsensical according to the git-fetch manpage:
> −−depth=&lt;depth&gt;
Limit fetching to the specified number of commits from the tip of each remote branch history. If fetching to a shallow repository created by git clone with −−depth=<depth> option (see git-clone(1)), deepen or shorten the history to the specified number of commits. Tags for the deepened commits are not fetched.

**Comment by matyasselmeci** [comment]
Is this repo going to be where the code for the workflow is kept, or is it just an experiment or something?

File: .github/workflows/validate-mkdocs.yml (hunk 1 above)
Comment (brianhlin) [reply .github/workflows/validate-mkdocs.yml:10]: Fetches all tags and branches: https://github.com/marketplace/actions/checkout#fetch-all-history-for-all-tags-and-branches

It technically only needs the most recent commits in `master` (for building the docs) and `gh-pages` (for deploying the docs) but I don't think there's an easy way to do that

**Comment by matyasselmeci** [comment]
Also I kinda want to clean up my link fixer before giving it to normal users...

**Comment by brianhlin** [comment]
I just want to get a review here + see it in action in the technology area first. In one of the earlier commits, I did get the deploy step working but not since : https://brianhlin.github.io/technology/

Don't worry about the link fixer, I suspect that we really only need it for the docs area.

File: .github/workflows/validate-mkdocs.yml (hunk 1 above)
Comment (matyasselmeci) [reply .github/workflows/validate-mkdocs.yml:10]: _0_ fetches everything??? Wow, GitHub really dropped the ball on the semantics here.

**Comment by matyasselmeci** [comment]
If we update the template, will we have to update all the individual repos as well?

**Comment by brianhlin** [comment]
Yup but that's not really all that different from our current submodule setup. The workflows are very simple so I don't think it's an issue to do mass updates if we need to in the future.

**Review by matyasselmeci — APPROVED** [review]

**Comment by brianhlin** [comment]
OTOH, if we do want to centralize updates for all repos in the future, that's a possibility:

1. Create GHAs reflecting the job steps in these workflows and version them
2. Modify the workflow templates to use the GHAs, specifying the major version
3. Make changes to the actions bumping the minor version

**Pull request merged**